lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -375,24 +380,47 @@ blockdev_nvme_dump_config_json(struct spdk_bdev *bdev, struct spdk_json_write_ct
 	spdk_json_write_name(w, "firmware_revision");
 	spdk_json_write_string(w, buf);
 
+	spdk_json_write_object_end(w);
+
+	spdk_json_write_name(w, "vs");
+	spdk_json_write_object_begin(w);
+
 	spdk_json_write_name(w, "nvme_version");
 	if (vs.bits.ter) {
 		spdk_json_write_string_fmt(w, "%u.%u.%u", vs.bits.mjr, vs.bits.mnr, vs.bits.ter);
 	} else {
 		spdk_json_write_string_fmt(w, "%u.%u", vs.bits.mjr, vs.bits.mnr);
 	}
 
-	spdk_json_write_name(w, "nsid");
+	spdk_json_write_object_end(w);
+
+	spdk_json_write_name(w, "csts");
+	spdk_json_write_object_begin(w);
+
+	spdk_json_write_name(w, "rdy");
+	spdk_json_write_uint32(w, csts.bits.rdy);
+
+	spdk_json_write_name(w, "cfs");
+	spdk_json_write_uint32(w, csts.bits.cfs);
+
+	spdk_json_write_object_end(w);
+
+	spdk_json_write_name(w, "ns_data");
+	spdk_json_write_object_begin(w);
+
+	spdk_json_write_name(w, "id");
 	spdk_json_write_uint32(w, spdk_nvme_ns_get_id(ns));
 
-	spdk_json_write_name(w, "ns_block_size");
+	spdk_json_write_name(w, "block_size");
```

Comment:
I didn't notice this on the previous review that introduced these fields, but block_size and total_size should be redundant with the data returned by the generic get_bdevs RPC (which this API is part of).  (Currently, total_size will not actually match the reported num_blocks * block_size because of the NvmeLunsPerNs splitting code, but we are working to remove that.)  We can keep them for now since they don't quite match up to the real namespace values (size is rounded down to a MB boundary), but I'd like to remove the duplication in the future unless there's a motivation for having these values distinct from the generic bdev ones.